### ISSUE TITLE
feat(MatSelect): Active state should take into account keyboard versus mouse interaction

### ISSUE THREAD

**Issue by elenachiosa** [angular/components#23953]
### Feature Description

Hello,
We would like to fully implement accessibility in terms of good color contrast and visible focus states ([WCAG Focus Visible](https://www.w3.org/WAI/WCAG21/Understanding/focus-visible.html)).
What we use right now to achieve this is the ":focus-visible" selector. We also use FocusMonitor from the A11yModule of CDK to create accessible keyboard interaction. FocusMonitor has a neat feature - it adds classes specific to the interaction type that the user uses (e.g. on keyboard interaction, the currently focused item has the ".cdk-keyboard-focus" class).

What I'd suggest as an improvement to the Select component is to have this difference illustrated in the component, out of the box. At the moment we are styling the ".mat-active" class in order to have a visible focus state for keyboard users but this looks very odd for mouse users.

### Use Case

Currently, the Select component looks exactly the same for both mouse and keyboard interaction.

What we would like to see in the future is the following:
Keyboard interaction with visible outline:
![image](https://user-images.githubusercontent.com/17402311/141339214-746b81b4-785f-45b8-a4a8-28c2d91d7dcb.png)
Mouse interaction:
![image](https://user-images.githubusercontent.com/17402311/141339235-d851d543-1c73-4226-ba57-4c90848e581d.png)

Stackblitz link with current behavior: https://stackblitz.com/edit/angular-7779kz?file=src%2Fapp%2Fselect-overview-example.html

**Comment by crisbeto** [comment]
It works like this because focus isn't actually on the `mat-option`, it's on the select's trigger. We use `aria-activedescendant` to manage focus.

**Comment by iKrishnaSahu** [comment]
Any updates on this feature?

**Comment by crisbeto** [comment]
We are closing this issue, because we've determined, using a semi-automated process, that it is no longer relevant, has low engagement, or it does not have enough information to proceed. Feel free to comment if you think the issue should be reopened, or open a new issue with the latest version of our code instead.
